diff --git a/compat/crypt.h b/compat/crypt.h
index 41c6454..ccdd83a 100644
--- a/compat/crypt.h
+++ b/compat/crypt.h
@@ -59,7 +59,7 @@ static int update_keys(unsigned long *pkeys, const z_crc_t *pcrc_32_tab, int c)
     (*(pkeys + 1)) += (*(pkeys + 0)) & 0xff;
     (*(pkeys + 1)) = (*(pkeys + 1)) * 134775813L + 1;
     {
-        register int keyshift = (int)((*(pkeys + 1)) >> 24);
+        int keyshift = (int)((*(pkeys + 1)) >> 24);
         (*(pkeys + 2)) = CRC32((*(pkeys + 2)), keyshift);
     }
     return c;
